esp/services/ws_logaccess/WsLogAccessService.cpp

```diff
@@ -0,0 +1,312 @@
+////
+////        ILogAccessFilter * andedcomponentsfilter = getBinaryLogAccessFilter(oredcomponentsfilter, right, LOGACCESS_FILTER_and);
+////
+////        andedcomponentsfilter->addToPTree(tree);
+////        StringBuffer xml;
+////        toXML(tree, xml);
+////        fprintf(stdout, "TTTTT%s", xml.str());
+////
+////        StringBuffer logcontent;
+////        fetchLog(logcontent, *m_esLogAccessor.get(), andedcomponentsfilter, range, cols);
+//        StringBuffer logcontent;
+//        fetchComponentLog(logcontent, *m_esLogAccessor.get(), componentname, range, cols);
+//        resp.setLog(logcontent.str());
+//    }
+//
+//    resp.setSuccess(success);
+//    if (message && *message)
+//        resp.setMessage(message);
+//
+//    return success;
+//}
+
+bool Cws_logaccessEx::onGetLogs(IEspContext &context, IEspGetLogsRequest &req, IEspGetLogsResponse & resp)
+{
+    const char * message = "success";
+    bool success = true;
+
+    CLogAccessType searchbycategory = req.getLogCategory();
+    const char * searchbyvalue = req.getValue();
+    if (searchbycategory != CLogAccessType_All && (!searchbyvalue || !*searchbyvalue))
+    {
+        message = "WsLogAccess::onGetLogs: Must provide log category";
+        success = false;
+    }
+    else
+    {
+        ILogAccessFilter * filter = nullptr;
+        switch (searchbycategory)
+        {
+            case CLogAccessType_All:
+                filter = getWildCardLogAccessFilter();
+                break;
+            case CLogAccessType_ByWUID:
+                filter = getWUIDLogAccessFilter(searchbyvalue);
+                break;
+            case CLogAccessType_ByComponent:
+                filter = getComponentLogAccessFilter(searchbyvalue);
+                break;
+            case CLogAccessType_ByLogType:
+                filter = getClassLogAccessFilter(LogMsgClass(LogMsgClassFromAbbrev(searchbyvalue)));
+                break;
+            case CLogAccessType_ByTargetAudience:
+                filter = getAudienceLogAccessFilter(MessageAudience(LogMsgAudFromAbbrev(searchbyvalue)));
+                break;
+            case LogAccessType_Undefined:
+            default:
+                break;
+        }
+
+        struct LogAccessTimeRange range = requestedRangeToLARange(req.getRange());
+        StringArray & cols = req.getColumns();
+
+        int limit = req.getLimit();
+        //if (limit < 0)
+        //    throw makeStringException(aud, code, why)
+        long offset = req.getOffset();
+
+        LogAccessConditions logFetchoptions;
+
+        logFetchoptions.timeRange = range;
+        logFetchoptions.filter = filter;
+        logFetchoptions.copyLogFieldNames(&cols);
+        logFetchoptions.limit = limit;
+        logFetchoptions.offset = offset;
+
+        StringBuffer logcontent;
+        m_esLogAccessor->fetchLog(logFetchoptions, logcontent);
+        //fetchLog(logcontent, *m_esLogAccessor.get(), filter, range, cols);
+        resp.setLog(logcontent.str());
```

Comment:
@jakesmith what jlib streaming interface would you envision using?
Also, the response buffer is currently json structured, as an array of "log" entries, which contains an array of "field" values (I'm adding a format option: XML/JSON/RAW).

I can add an array of field arrays interface if it would be more convenient to the caller, what's your thought on the usefulness of that version vs the xml/json buffer? 
Pseudo xml representation of the log output:
`<logentries><logentry><fields><timestamp>...</timestamp><message>...</message>..</fields></logentry>...</logentries>`

Reply:
given ECM format limitations, I suspect a string is reasonable.
I hadn't realized it was json encoded as a string though , a better field name and/or comment would help clarify.
As discussed offline a request format (default json perhaps) would be a nice to have, giving the requester control over the format.
Perhaps where one format is CSV , so can get plain text back, which for ZAP type reports is probably the format of choice.